### PR TITLE
Adding CRAN package 'semaphore'

### PR DESCRIPTION
**Pull request by dansmith01**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by dansmith01** [comment]
I am listing myself as the maintainer. Recipe built using github.com/bgruening/conda_r_skeleton_helper. Checklist completed.

@conda-forge/help-r Ready for review!

File: recipes/r-semaphore/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    - {{ posix }}zip               # [win]
+    - cross-r-base {{ r_base }}    # [build_platform != target_platform]
+  host:
+    - r-base
+    - r-bh
+    - r-rcpp
+  run:
+    - r-base
+    - {{ native }}gcc-libs         # [win]
+    - r-bh
```
Comment (cbrueffer) [inline recipes/r-semaphore/meta.yaml:46]:
```suggestion
    #- r-bh
```

`r-bh` is quite big and only needed during compile-time.

**Review by cbrueffer — CHANGES_REQUESTED** [review]
Looks good; only one suggestion, then it's ready to go.

**Comment by dansmith01** [comment]
@cbrueffer clever idea! Seems to break the build on OSX though. Revert the changes?

**Comment by cbrueffer** [comment]
The error is unrelated to the change; there are general macOS build issues at the moment: https://github.com/conda-forge/staged-recipes/issues/28242

**Comment by dansmith01** [comment]
Thanks @cbrueffer !

**Pull request merged**